source/simulator/core.cc

```diff
@@ -905,7 +905,8 @@ namespace aspect
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
             }
         }
-      coupling[x.temperature][x.temperature] = DoFTools::always;
+      if (!(parameters.nonlinear_solver == NonlinearSolver::Kind::no_Advection_iterated_Stokes))
```

Comment:
can you leave a comment? Maybe something like "Do not allocate a temperature matrix if no temperature solves are going to be performed"?

Reply:
Looks good, but could you also include `NonlinearSolver::Kind::no_Advection_no_Stokes`? I know that one would need additional adjustments elsewhere, but lets start here. Otherwise ready.

Reply:
Done.